persistence/relational-jdbc/src/main/java/org/apache/polaris/persistence/relational/jdbc/DatasourceOperations.java

```diff
@@ -200,6 +202,61 @@ public int executeUpdate(QueryGenerator.PreparedQuery preparedQuery) throws SQLE
         });
   }
 
+  /**
+   * Executes the INSERT/UPDATE Queries in batches. Requires that all SQL queries have the same
+   * parameterized form.
+   *
+   * @param preparedQueries : queries to be executed
+   * @return : Number of rows modified / inserted.
+   * @throws SQLException : Exception during Query Execution.
+   */
+  public int executeBatchUpdate(List<QueryGenerator.PreparedQuery> preparedQueries)
+      throws SQLException {
+    if (preparedQueries.isEmpty()) {
+      return 0;
+    }
+    int batchSize = 100;
+    AtomicInteger successCount = new AtomicInteger();
+    return withRetries(
+        () -> {
+          String sql = preparedQueries.get(0).sql();
```

Comment:
Am I correct that this is relying on the assumption that all `PreparedQuery` instances have the exact same SQL query?

The issue is that nothing in the method signature prevents a caller from calling this method with different SQL queries. I think at the very least, you should add a check for this invariant. But this hints at a badly designed signature. A proper signature would be something like:

```java
public int executeBatchUpdate(QueryGenerator.PreparedBatchQuery preparedBatchQuery)
```